Added user_categories migration, model, endpoint

green-arrow opened the pull request:
Addresses #39

scarfdestroyer commented:
Potentially satisfying issue https://github.com/code-corps/code-corps-api/issues/83

JoshSmith commented on spec/workers/add_profile_picture_from_gravatar_worker_spec.rb:
```diff
@@ -0,0 +1,28 @@
+  end
+
+  context "when the user does not have a gravatar", vcr: { cassette_name: 'workers/add gravatar image worker/adds gravatar' } do
+    it "a photo has not been added" do
```
These `it` blocks need re-written to read well.

JoshSmith commented:
That change should be all that's needed and then :shipit: :+1:

JoshSmith commented:
Your its don't read the same. "adds the photo"/"does not add the photo" context-switches less.

JoshSmith commented:
@scarfdestroyer update the specs when you get a moment and this will get merged.

green-arrow committed:
Added user_categories migration, model, endpoint

Addresses #39

joshsmith commented on priv/repo/migrations/20160822020401_create_user_category.exs:
```diff
@@ -0,0 +1,14 @@
+      timestamps()
+    end
+
+    create index(:user_categories, [:user_id, :category_id], unique: true, name: :index_projects_on_user_id_category_id)
```
Same point on name here as before.

joshsmith commented on test/controllers/user_category_controller_test.exs:
```diff
@@ -0,0 +1,112 @@
+  use CodeCorps.ConnCase
+
+  alias CodeCorps.UserCategory
+  alias CodeCorps.Repo
```
Would reorder alpha.

joshsmith commented:
Same point here on a `UserCategoryViewTest`.

I would also consider doing a `UserViewTest` and ensuring that the `has_many` categories through is present there.

Otherwise looks great!

green-arrow commented:
@JoshSmith - If we can get the user roles branch merged, I'll have a basis to add on to the `UserViewTest` I added there :)

green-arrow committed:
Addressing code review comments.

Added user_categories_view_test
Added categories relation to user model
Preloaded categories in all user endpoints

green-arrow commented:
@JoshSmith - Addressed the code review comments. My computer is getting ready to die and I have no idea where my charger is, so feel free to squash and merge (have to hit `Merge pull request` first to get that option) if there are no outstanding issues. If there are, I'll make sure to get them when I wake up in the morning and find my charger :)

joshsmith committed:
Address code review comments

joshsmith commented:
👍 Looks great, thanks @green-arrow!

joshsmith committed:
Fix typo
